### PR TITLE
[PRM-797]:  'You have uploaded X file(s) page'

### PR DESCRIPTION
**Pull request by BackRowOla**

### CODE REVIEW

**Comment by LewisMitchell** [comment]
Is it possible you can rebase this instead of pullin in main? It's introducing changes that are already in main

File: test/controllers/OtherReasonControllerSpec.scala [hunk 1]
```diff
@@ -587,16 +563,18 @@ class OtherReasonControllerSpec extends SpecBase {
...
-            .thenReturn(Future.successful(5))
+          val s1 = UploadJourney("ref1",UploadStatusEnum.READY,None,None,None,LocalDateTime.now())
+          when(mockUploadJourneyRepository.getUploadsForJourney(any())).thenReturn(Future.successful(Some(Seq(s1,s1,s1,s1,s1))))
+
           val result: Future[Result] = controller.onSubmitForUploadComplete(NormalMode)(fakeRequestConverter(fakeRequestWithCorrectKeys.withJsonBody(
             Json.parse(
               """
                 |{
-                | "value": ""
+                | "value": "yes"
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:573]: This should be ""
```suggestion
                | "value": ""
```

File: test/controllers/OtherReasonControllerSpec.scala [hunk 2]
```diff
@@ -449,9 +427,7 @@ class OtherReasonControllerSpec extends SpecBase {
 
     "removeFileUpload" should {
       "the user is authorised" must {
-        "redirect to multi-file upload page when the JS enabled cookie is present and the routing feature switch is off" in new Setup(AuthTestModels.successfulAuthResult) {
-          when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
-            .thenReturn(false)
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:454]: Could you add this back?

File: test/controllers/OtherReasonControllerSpec.scala [hunk 3]
```diff
@@ -325,23 +315,11 @@ class OtherReasonControllerSpec extends SpecBase {
...
-          val result: Future[Result] = controller.onPageLoadForFirstFileUpload(NormalMode)(userRequestWithCorrectKeys)
-          status(result) shouldBe OK
-        }
-
-        "return 303 (SEE_OTHER) when the feature switch is disabled" in new Setup(AuthTestModels.successfulAuthResult) {
-          when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
-            .thenReturn(false)
-          val result: Future[Result] = controller.onPageLoadForFirstFileUpload(NormalMode)(userRequestWithCorrectKeysAndJS)
-          status(result) shouldBe SEE_OTHER
-        }
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:344]: Could you add this back?

File: test/controllers/OtherReasonControllerSpec.scala [hunk 4]
```diff
@@ -215,33 +218,20 @@ class OtherReasonControllerSpec extends SpecBase {
...
-        "return OK when the feature switch is disabled" in new Setup(AuthTestModels.successfulAuthResult) {
-          when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
-            .thenReturn(false)
-          val result: Future[Result] = controller.onPageLoadForUploadEvidence(NormalMode)(userRequestWithCorrectKeysAndJS)
-          status(result) shouldBe OK
-        }
-
-        "return 303 (SEE_OTHER) when the feature switch is enabled" in new Setup(AuthTestModels.successfulAuthResult) {
-          when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
-            .thenReturn(true)
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:244]: Could you add this back?

File: test/controllers/OtherReasonControllerSpec.scala [hunk 5]
```diff
@@ -215,33 +218,20 @@ class OtherReasonControllerSpec extends SpecBase {
...
-        when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
-          .thenReturn(false)
         val result: Future[Result] = controller.onPageLoadForUploadEvidence(NormalMode)(userRequestWithCorrectKeysAndJS)
         status(result) shouldBe OK
       }
 
       "return OK and correct view (pre-populated date when present in session)" in new Setup(
         AuthTestModels.successfulAuthResult, Some(UploadData.oneWaitingUploads)
       ) {
-        when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:227]: Could you add this back?

File: test/controllers/OtherReasonControllerSpec.scala [hunk 6]
```diff
@@ -215,33 +218,20 @@ class OtherReasonControllerSpec extends SpecBase {
     "onPageLoadForUploadEvidence" when {
 
       "return OK and correct view" in new Setup(AuthTestModels.successfulAuthResult) {
-        when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:218]: Could you add this back?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Comment by ruthdas03** [comment]
> Is it possible you can rebase this instead of pulling in main? It's introducing changes that are already in main

rebased it .

**Comment by LewisMitchell** [comment]
Thanks for the rebase, just some further changes and then I can do some testing.

**Comment by ruthdas03** [comment]
> Thanks for the rebase, just some further changes and then I can do some testing.

Added them back and fixed the test case.

File: test/controllers/OtherReasonControllerSpec.scala [hunk 7]
```diff
@@ -242,6 +243,8 @@ class OtherReasonControllerSpec extends SpecBase {
         "return 303 (SEE_OTHER) when the feature switch is enabled" in new Setup(AuthTestModels.successfulAuthResult) {
           when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
             .thenReturn(true)
+        }
+        "return 303 (SEE_OTHER)" in new Setup(AuthTestModels.successfulAuthResult) {
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:247]: We can remove this it should be

```
        "return 303 (SEE_OTHER) when the feature switch is enabled" in new Setup(AuthTestModels.successfulAuthResult) {
          when(mockFeatureSwitching.isEnabled(ArgumentMatchers.any()))
            .thenReturn(true)
          val result: Future[Result] = controller.onPageLoadForUploadEvidence(NormalMode)(userRequestWithCorrectKeys)
          status(result) shouldBe SEE_OTHER
        }
```

File: test/controllers/OtherReasonControllerSpec.scala [hunk 8]
```diff
@@ -325,23 +328,29 @@ class OtherReasonControllerSpec extends SpecBase {
...
+          .thenReturn(false)
+        val result: Future[Result] = controller.onPageLoadForFirstFileUpload(NormalMode)(userRequestWithCorrectKeysAndJS)
+        status(result) shouldBe SEE_OTHER
+      }
+
+      "return SEE_OTHER and correct view when JS is enabled" in new Setup(AuthTestModels.successfulAuthResult) {
+        when(mockUpscanService.initiateSynchronousCallToUpscan(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any(), ArgumentMatchers.any()))
+          .thenReturn(Future.successful(Right(UpscanInitiateResponseModel("file1ref", UploadFormTemplateRequest("/", Map.empty)))))
+        val result: Future[Result] = controller.onPageLoadForFirstFileUpload(NormalMode)(userRequestWithCorrectKeysAndJS)
+        status(result) shouldBe SEE_OTHER
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:353]: This test can be removed

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just some final changes

**Comment by ruthdas03** [comment]
Updated with the changes requested.

**Pull request merged**